os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -173,23 +174,17 @@ void binary_manager_register_kpart(int part_num, int part_size)
 void binary_manager_update_running_state(int bin_id)
 {
 	int bin_idx;
+	struct tcb_s *tcb;
 
-	if (bin_id <= 0) {
-		bmdbg("Invalid parameter: bin id %d\n", bin_id);
-		return;
-	}
+	tcb = sched_gettcb(bin_id);
```

Comment:
I guess this is very important function which is change state, but it never return error case...
You should check some error case (at lease EINVAL is necessary, I think)
bin_id is always valid? Yes this will work internally but someone can open BINMGR_REQUEST_MQ